controller/src/main.cpp

```diff
@@ -117,39 +159,23 @@ static void controller_loop() {
     DEV_MODE_comms_handler(controller_status, &gui_status);
```

Comment:
Pretty much the whole body of this while() loop has to be under a BlockInterrupts, right? Cause the high-priority task reads and writes elements of ControllerStatus, and so does the body of this loop.

Reply:
No, we don't want to block interrupts during the whole comm handler because that pretty much brings us back to where we are now, the jitter depends on the time spend in the comm handler which is unpredictable.
The comm handler should update a local data structure which we'll copy over quickly with interrupts disabled.

Reply:
Then we need to block interrupts and make a local copy of controller_status before calling the comms handler on the copy. Otherwise the comms handler can have the high-priority loop change the controller status while it's being read.